models/marketing/hubspot__contacts.sql

```diff
@@ -54,29 +55,39 @@ with contacts as (
     select 
         {{ cte_ref }}.*,
         {% for metric in engagement_metrics() %}
-        coalesce(engagements.{{ metric }},0) as {{ metric }} {% if not loop.last %},{% endif %}
+        coalesce(engagements.{{ metric }},0) as {{ metric }}{{ ',' if not loop.last }}
         {% endfor %}
     from {{ cte_ref }}
     left join engagements
-        using (contact_id)
-
-)
+        on {{ cte_ref }}.contact_id = engagements.contact_id
 
-select *
-from engagements_joined
+{% set cte_ref = 'engagements_joined' %}
+{% endif %}
 
-{% elif emails_enabled %}
+{% if forms_enabled %}
+), conversions as (
 
-)
+    select *
+    from {{ ref('int_hubspot__form_metrics__by_contact') }}
 
-select *
-from email_joined
+), conversions_joined as (
 
-{% else %}
+    select 
+        {{ cte_ref }}.*,
+        {% for first_or_last in ['first_conversion', 'most_recent_conversion']%}
+            {% for metric in ['form_id', 'date', 'form_name', 'form_type'] %}
+                conversions.{{first_or_last}}_{{ metric }}, -- add these for emails & engagements if time, otherwise feature request
```

Comment:
Let's remove this comment and create the FR

Reply:
Removed.